client/blocks/eligibility-warnings/index.jsx

```diff
@@ -2,139 +2,85 @@
+	isEligible,
+	isJetpack,
+	isPlaceholder,
+	onProceed,
+	siteId,
+	siteSlug,
+	translate,
+} ) => {
+	const context = -1 !== backUrl.indexOf( 'plugins' ) ? 'plugins' : 'themes';
 
-const EligibilityWarnings = props => {
-	const {
-		translate,
-		backUrl,
-		onProceed,
-		siteId,
-		isEligible,
-		eligibilityData,
-		isPlaceholder,
-	} = props;
-
-	const holdMessages = getHoldMessages( translate );
-	const holds = get( eligibilityData, 'eligibilityHolds', [ 'PLACEHOLDER', 'PLACEHOLDER' ] );
 	const warnings = get( eligibilityData, 'eligibilityWarnings', [] );
 
-	const classes = classNames( {
-		'eligibility-warnings__message': true,
-		'eligibility-warnings__placeholder': isPlaceholder,
-	} );
+	let holds = get( eligibilityData, 'eligibilityHolds', [ 'PLACEHOLDER', 'PLACEHOLDER' ] );
+	const bannerHolds = filter( holds, hold => -1 !== [ 'NO_BUSINESS_PLAN', 'NOT_USING_CUSTOM_DOMAIN' ].indexOf( hold ) );
+	holds = difference( holds, bannerHolds );
+
+	const classes = classNames(
+		'eligibility-warnings',
+		{ 'eligibility-warnings__placeholder': isPlaceholder }
+	);
 
 	return (
-		<div className="eligibility-warnings">
+		<div className={ classes }>
 			<QueryEligibility siteId={ siteId } />
-			<SectionHeader label={ translate( 'Conflicts' ) } />
 
-			{ map( holds, ( error, index ) =>
-				<Card key={ index } className={ classes }>
-					<Gridicon icon="notice" className="eligibility-warnings__error-icon" />
-					<div className="eligibility-warnings__message-content">
-						<div className="eligibility-warnings__message-title">
-							{ translate( 'Error: %(title)s', { args: { title: holdMessages[ error ].title } } ) }
-						</div>
-						<div className="eligibility-warnings__message-description">
-							{ holdMessages[ error ].description }
-						</div>
-					</div>
-					<div className="eligibility-warnings__message-action">
-						<Button href={ holdMessages[ error ].supportUrl } target="_blank" rel="noopener noreferrer">
-							{ translate( 'Resolve' ) }
-						</Button>
-					</div>
-				</Card>
-			) }
+			{ 'plugins' === context && ! hasBusinessPlan && ! isJetpack &&
+				<Banner
+					description={ translate( 'Please upgrade to install this plugin.' ) }
+					feature={ FEATURE_UPLOAD_PLUGINS }
+					plan={ PLAN_BUSINESS }
+					title={ translate( 'Business plan required' ) }
+				/>
+			}
+			{ 'themes' === context && ! hasBusinessPlan && ! isJetpack &&
+				<Banner
+					description={ translate( 'Unlimited themes, advanced customization, no ads, live chat support, and more!' ) }
+					feature={ FEATURE_UPLOAD_THEMES }
+					plan={ PLAN_BUSINESS }
+					title={ translate( 'To upload themes, upgrade to Business Plan' ) }
+				/>
+			}
+			{ hasBusinessPlan && ! isJetpack && includes( bannerHolds, 'NOT_USING_CUSTOM_DOMAIN' ) &&
+				<Banner
+					className="eligibility-warnings__banner"
+					description={ translate( 'Add a free custom domain to install this plugin.' ) }
+					href={ `/domains/add/${ siteSlug }` }
```

Comment:
@roundhill If you don't have a custom domain, `/domains/manage` brings you to a page with a big blue notice saying "Free domain available - Claim Free Domain", and clicking on it - which would be the action required to solve the AT hold - opens `/domains/add`, where you can pick your custom domain of choice for free.

So I'd say we should keep `/domains/add` here, in order to save the user one click and a bit of confusion.